### DISCUSSION TITLE
Best practice to handle attachments/images etc.

### DISCUSSION THREAD

**Discussion by ghost** [Zettlr/Zettlr#3216] [Q&A]
Dear all,

what is your way of storing images and similar things you link to in your markdown files?

Everything into a dedicated folder? Keep it with the markdown file in it's folder? Relative paths or absolute paths?

The Zettelkasten approach would put everything into one and only one folder. But I still want to have some way of structure by using folders.

Using more than one folder, additional work may be needed if one decides to split up a folder's content, or lift/lower the markdown file to another directory level. In this case and using a dedicated folder for images and the like with absolute path declaration in the markdown file will keep everything fine. But what if you sync your notes with another device where the file structure is different?

In case of relative paths one has to adjust all path declarations, which is often anoying because of rather cryptic and hard to remember file names.

I was trying to use soft links, but this may make/makes (in case of using syncthing) problems. Moreover soft links are not supported on android devices (AFAIK).

Very much looking forward to your thoughts! Thanks a lot in advance!

Markus

**Comment by nathanlesage** [comment]
The default values for preferences in Zettlr can give you a hint as to how I do it. Normally, I store images in a relative folder `assets` which is relative to the current file. However, I don't use that many images. When I move files, I can just take the `assets` folder with me.

However, I would state that moving files around is only something you'd do early on, but later moving files around is not the issue anymore. So I wouldn't bother too much about broken links to images. Furthermore, even if a link breaks, given the filename it will be easy to find that file again, so even then it won't be a big problem!

Then (again, this is my subjective way of working so it will not work 1:1 for you!), I have come to the following (coarse) directory structure:

* 1 folder for my PhD. In it I store everything written with regard to that, and create new folders as needed, but I don't try to adhere to a specific system. Rather, I use whatever I need when I need it
* 1 folder for my Zettelkasten. In that I store all my "atomic" files.
* 1 folder for reading notes. I personally differentiate between Zettelkasten and reading notes, and this has proven valuable for me.
* All my data analysis, however, **is not even loaded in Zettlr**, so I keep that completely separate, because the problems you sometimes have with code are completely orthogonal to problems you have with your texts

So TL;DR: Don't worry about broken links or moving files, just get started. The very worst that could happen is that you have to spend one day reorganizing everything. Not ideal, yes, but certainly manageable! Find something that works *for you*, not something other people tell you to do!

**Comment by ghost** [comment]
Thanks @nathanlesage for your detailed thoughts!

I am sure that at some point I will have settled on a workflow, that suits me best. However, right now I am just wondering, whether I have overlooked something that could be at least worth trying or not seen certain pitfalls, which I could avoid then right from the beginning.

**Comment by leo-hstone** [comment]
I also had some thoughts on this topic. I use Zettlr as notetaking app with a folder structure. Every image I use, I put in the same folder as the file with a relative path.

I had to move a few files recently and just took the pictures manually with them. I know this is not optimal, but it works for me as I don't have to restructure that often. The next time I do a major restructuring I want to write a script that automatically moves linked assets. Some skilled programmer could achieve this with a oneliner I guess 😄
